homeassistant/components/vesync/config_flow.py

```diff
@@ -0,0 +1,66 @@
+"""Config flow utilities."""
+import logging
+from collections import OrderedDict
+import voluptuous as vol
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_instances(hass):
+    """Return already configured instances."""
+    return [
+        entry.data[CONF_USERNAME]
+        for entry in hass.config_entries.async_entries(DOMAIN)]
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class VeSyncFlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Instantiate config flow."""
+        self._username = None
+        self._password = None
+        self._time_zone = None
+        self.data_schema = OrderedDict()
+        self.data_schema[vol.Required(CONF_USERNAME)] = str
+        self.data_schema[vol.Required(CONF_PASSWORD)] = str
+
+    async def _show_form(self, errors=None):
+        """Show form to the user."""
+        return self.async_show_form(
+            step_id='user',
+            data_schema=vol.Schema(self.data_schema),
+            errors=errors if errors else {},
+        )
+
+    async def async_step_import(self, import_config):
+        """Handle external yaml configuration."""
+        return await self.async_step_user(import_config)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        if configured_instances(self.hass):
+            return self.async_abort(reason='already_setup')
+
+        if not user_input:
+            return await self._show_form()
+
+        self._username = user_input[CONF_USERNAME]
+        self._password = user_input[CONF_PASSWORD]
+
```

Comment:
We should create a manager here and check that credentials are correct.

Reply:
I tried to create the manager and login here but when I try to store manager I get a Key Error for `self.hass.data[DOMAIN]` when trying to create a manager key even though I set `self.hass.data[DOMAIN] = {}`  in async_setup().  And if I tried to put the manager in the config entry I get a json deserialization error.  What am I missing?

Reply:
We can't store non serializable objects, like the manager, in the data for the config entry, since it will be saved in a json file on disk. Create a manager here to check credentials, then create a new manager in `async_setup_entry`.

The config flow is started before the component is set up, if triggered by the user in the GUI.